codelists/actions.py

```diff
@@ -832,34 +832,68 @@ def fork_codelist(codelist, new_owner):
     forked_version.id = None
     forked_version.save()
 
-    def deep_copy_related(related):
-        related = deepcopy(related)
-        version_field = None
-        for related_model_field in related.__class__._meta.get_fields():
-            if (
-                hasattr(related_model_field, "related_model")
-                and related_model_field.related_model == CodelistVersion
-            ):
-                version_field = related_model_field
-                break
-        if not version_field:
-            raise AttributeError(
-                f"Unable to locate version field for {related.__class__} model"
-            )
-        setattr(related, version_field.name, forked_version)
-        related.id = None
-        related.save()
-
-    for field in CodelistVersion._meta.get_fields():
-        match field.__class__:
-            case reverse_related.OneToOneRel:
-                if hasattr(latest_version, field.name):
-                    related = getattr(latest_version, field.name)
-                    deep_copy_related(related)
-            case reverse_related.ManyToOneRel:
-                if hasattr(latest_version, field.name):
-                    for related in getattr(latest_version, field.name).all():
-                        deep_copy_related(related)
-            case reverse_related.ManyToManyRel:
-                raise NotImplementedError
+    forked_related_objects = {}
+
+    def deep_copy_related_fields(source, destination):
```

Comment:
This looks good - but it's pretty complicated and I'm wondering if there may be unconsidered edge cases. If you're ok to own it I'm happy to go with it, but an alternative might be to explicitly copy all the relations that currently exist, and then add a test that fails if new relations are added to the model.

Reply:
I'm happy to own this, the recursion is a bit ungainly due to the need to save the parent with a new ID before iterating the children but otherwise I'm happy. The tests give me enough confidence to proceed